src/NetAnalyzers/Core/Microsoft.NetCore.Analyzers/Performance/UseCountProperly.cs

```diff
@@ -314,6 +321,48 @@ private static bool AnalyzeParentInvocationOperation(IInvocationOperation parent
             return true;
         }
 
+        private static bool AnalyzeIsPatternOperation(IIsPatternOperation isPatternOperation, out bool shouldNegateIsEmpty)
+        {
+            shouldNegateIsEmpty = true;
+
+            if (isPatternOperation.Pattern is INegatedPatternOperation negatedPattern)
+            {
+                if (negatedPattern.Pattern is IRelationalPatternOperation { OperatorKind: BinaryOperatorKind.GreaterThan, Value: ILiteralOperation { ConstantValue: { HasValue: true, Value: 0 } } }
```

Comment:
I would suggest adding comments for each of these cases where you have the code corresponding to the condition for easy understanding, probably you can just copy over the corresponding test case code that you have added for each of the conditional branches in this method.